stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -681,14 +701,17 @@ extension TaskGroup: AsyncSequence {
     return Iterator(group: self)
   }
 
-  /// Allows iterating over results of tasks added to the group.
+  /// A type that provides an iteration interface
+  /// over the results of tasks added to the group.
   ///
-  /// The order of elements returned by this iterator is the same as manually
-  /// invoking the `group.next()` function in a loop, meaning that results
-  /// are returned in *completion order*.
+  /// The elements returned by this iterator
+  /// appear in the order that the tasks *completed*,
+  /// not in the order that those tasks were added to the task group.
   ///
-  /// This iterator terminates after all tasks have completed successfully, or
-  /// after any task completes by throwing an error.
+  /// This iterator terminates after all tasks have completed successfully,
+  /// or after any task completes by throwing an error.
+  /// If a task completes by throwing an error,
+  /// no further task results are returned.
```

Comment:
This wording is tricky.

This is legal:

```
group.spawn { 1 }
group.spawn { throw boom }
group.spawn { 2 }

do { 
  for await t in group { 
    print(t)
  }
} catch { just ignore the error } 

// wait again on tasks
for await t in group { 
  print(t)
}
```

for this snippet it is legal to return: 1 (1st loop), error, 2 (2nd loop); there's numerous combinations in which this may complete but ths one is legal too.

So it's important to explain that the _iterator made by the AsyncSequence_ will stop returning after something throws -- thats what makes the async loop and manual use of the iterator equivalent. But one is allowed to try waiting again